src/api/auks/auks_krb5_cred.c

```diff
@@ -124,6 +124,111 @@ auks_krb_is_local_tgt(krb5_principal princ, krb5_data *realm)
 		auks_krb_data_eq(princ->data[1], *realm);
 }
 
+int
+auks_krb_cc_new_unique(char ** fullname_out)
+{
+        int fstatus;
+
+        krb5_context context;
+	krb5_ccache ccache;
+
+	char *ccache_type = NULL;
+
+	/* Initialize KRB5 context */
+	fstatus = krb5_init_context(&context);
+	if (fstatus) {
+		auks_error("Error while initializing KRB5");
+		fstatus = AUKS_ERROR_KRB5_CRED_INIT_CTX;
+		return fstatus;
+	}
+
+	/* Get the default ccache */
+	fstatus = krb5_cc_default(context, &ccache);
+	if (fstatus) {
+	        auks_error("Error while getting default ccache");
+		fstatus = AUKS_ERROR_KRB5_CRED_OPEN_CC ;
+		goto out;
+	}
+
+	/* Get the default ccache type */
+	ccache_type = krb5_cc_get_type(context, ccache);
+
+	/* Generate a new unique ccache */
+	fstatus = krb5_cc_new_unique(context, ccache_type, NULL, &ccache);
+	if (fstatus) {
+		auks_error("Error while generating new unique ccache of type %s", ccache_type);
+		fstatus = AUKS_ERROR_KRB5_CRED_OPEN_CC ;
+		goto out;
+	}
+
+
+	fstatus = krb5_cc_get_full_name(context, ccache, fullname_out);
+	if (fstatus) {
+		auks_error("Error while geting ccache full name");
+		fstatus = AUKS_ERROR_KRB5_CRED_OPEN_CC ;
+		goto out;
+	}
+
+
+	/* Call krb5_cc_switch for ccache that supports if */
+	if (krb5_cc_support_switch(context, ccache_type)) {
+		fstatus = krb5_cc_switch(context, ccache);
+		if (fstatus) {
+			auks_error("Error while calling krb5_cc_switch");
+			fstatus = AUKS_ERROR_KRB5_CRED_OPEN_CC ;
+			goto out;
+		}
+	}
```

Comment:
this call will have an impact on all the users sessions on the system that will not clearly specify a particular ccache to use using the KRB5CCNAME env variable. I do not think that it is interesting/usefull to put that here.
The goal of the function is to create a new unique ccache for a particular usage, not necessarily to switch all the user session to that ccache.
If this second purpose is required, it should be put in a second function, but I do not think it is necessary for what you have in mind.
The previous comment referring this one correspond to things that could also be removed with that block.

Reply:
In theory it seems like that. It should not be done. But in the case that the newly created ccache in the only one present, without that call, the object representing the default ccache of the collection is not created.
That makes me think that it only works when you have that particular environment variable. 
ie. It this creates the KCM:%{euid}:XXXX ccache, I do not think that KCM:%{euid} points to something usable.
This is clearly a corner case. Maybe, this call make the whole thing work if you don't have that `KRB5CCNAME` variable.
OpenSSH already does that, ccache are set as default on each and every new session. It creates a new one and call `krb5_cc_switch` onto it.

Reply:
On the other hand, if the session have a `KRB5CCNAME` pointing to a valid ccache, the default ccache of the collection is never used.
Thus, if all sessions of a user have a `KRB5CCNAME` env variable set, this is useless.